pandas/util/_decorators.py

```diff
@@ -19,46 +19,42 @@ def deprecate(name, alternative, version, alt_name=None,
 
     Parameters
     ----------
-    name : str
-        Name of function to deprecate
-    alternative : str
-        Name of function to use instead
+    name : func
+        Function to deprecate.
+    alternative : func
+        Function to use instead.
     version : str
-        Version of pandas in which the method has been deprecated
+        Version of pandas in which the method has been deprecated.
     alt_name : str, optional
-        Name to use in preference of alternative.__name__
+        Name to use in preference of alternative.__name__.
     klass : Warning, default FutureWarning
     stacklevel : int, default 2
     msg : str
-          The message to display in the warning.
-          Default is '{name} is deprecated. Use {alt_name} instead.'
+        The message to display in the warning.
+        Default is '{name} is deprecated. Use {alt_name} instead.'
     """
 
     alt_name = alt_name or alternative.__name__
     klass = klass or FutureWarning
     warning_msg = msg or '{} is deprecated, use {} instead'.format(name,
                                                                    alt_name)
 
+    # adding deprecated directive to the docstring
+    msg = msg or 'Use `{alt_name}` instead.'.format(alt_name=alt_name)
+
+    @Substitution(version=version, msg=msg)
+    @Appender(name.__doc__)
     @wraps(alternative)
     def wrapper(*args, **kwargs):
+        """
+        .. deprecated:: %(version)s
+
+           %(msg)s
+
+        """
```

Comment:
I am not sure why, but it seems this does not give the correct docstring. 
Can you check that `pd.Series.argmin?` still gives the deprecation warning at the top? (I checked out this branch and didn't see it, while on master it is there)